src/mscorlib/src/System/AppDomain.cs

```diff
@@ -590,37 +579,18 @@ public Object GetData(string name)
                 throw new ArgumentNullException(nameof(name));
             Contract.EndContractBlock();
 
-            int key = AppDomainSetup.Locate(name);
-            if (key == -1)
-            {
-                if (name.Equals(AppDomainSetup.LoaderOptimizationKey))
-                    return FusionStore.LoaderOptimization;
-                else
-                {
-                    object data;
-                    lock (((ICollection)LocalStore).SyncRoot)
-                    {
-                        LocalStore.TryGetValue(name, out data);
-                    }
-                    if (data == null)
-                        return null;
-                    return data;
-                }
-            }
+            if (name.Equals(AppDomainSetup.LoaderOptimizationKey))
```

Comment:
This should be removed as well. (LoaderOptimization is not used for anything in CoreCLR.)